core/services/gateway/connector/connector.go

```diff
@@ -76,8 +77,8 @@ type gatewayState struct {
 	wsClient network.WebSocketClient
 }
 
-func NewGatewayConnector(config *ConnectorConfig, signer Signer, handler GatewayConnectorHandler, clock clockwork.Clock, lggr logger.Logger) (GatewayConnector, error) {
-	if config == nil || signer == nil || handler == nil || clock == nil || lggr == nil {
+func NewGatewayConnector(config *ConnectorConfig, signer Signer, clock clockwork.Clock, lggr logger.Logger) (GatewayConnector, error) {
```

Comment:
why not give it an array of handlers?

Reply:
We don't have them when the connector is constructed.

Reply:
If we had a single handler before for calling the connector, why wouldn't we have one still?  Is it because the service_wrapper is being created and it won't be ready at connector construction time?

Reply:
We had one handler ready here before only because in Functions the Connector object was created by a job spec. Now that you moved it to be a core service, it's not the case any more. Core services (now including Gateway Connector) are all created first, before iterating over job specs and creating services that they need. In the re-design proposal (see doc I shared earlier today), each standard capability will add its own connector handler, when necesary.